test/libsolidity/syntaxTests/viewPureChecker/write_storage_fail.sol

```diff
@@ -3,4 +3,4 @@ contract C {
     function f() view public { x = 2; }
 }
 // ----
-// Warning: (56-57): Function declared as view, but this expression (potentially) modifies the state and thus requires non-payable (the default) or payable.
```

Comment:
Shouldn't there be more tests?

Reply:
It surprised me as well, but this seems to be the only one at the moment.

Reply:
Should I just add more tests in this PR?

Reply:
Yes, please.